// @aws-amplify/ui-angular/esm2015/lib/primitives/text-field/text-field.component.js
import { Component, HostBinding, Input } from '@angular/core';
import { nanoid } from 'nanoid';
export class TextFieldComponent {
    constructor() {
        this.autocomplete = 'new-password';
        this.disabled = false;
        this.fieldId = `amplify-field-${nanoid(12)}`;
        this.initialValue = '';
        this.label = '';
        this.placeholder = '';
        this.required = true;
        this.labelHidden = false;
        this.display = 'contents';
    }
}
TextFieldComponent.decorators = [
    { type: Component, args: [{
                selector: 'amplify-text-field',
                template: "<label\n  class=\"amplify-label\"\n  [class.amplify-visually-hidden]=\"labelHidden\"\n  [for]=\"fieldId\"\n>\n  {{ label }}\n</label>\n<input\n  class=\"amplify-input\"\n  [id]=\"fieldId\"\n  [type]=\"type\"\n  [name]=\"name\"\n  [placeholder]=\"placeholder\"\n  [required]=\"required\"\n  [value]=\"initialValue\"\n  [autocomplete]=\"autocomplete\"\n  [attr.disabled]=\"disabled ? '' : null\"\n  [attr.aria-invalid]=\"hasError ? 'true' : 'false'\"\n  [attr.aria-describedby]=\"describedBy\"\n/>\n"
            },] }
];
TextFieldComponent.propDecorators = {
    autocomplete: [{ type: Input }],
    disabled: [{ type: Input }],
    fieldId: [{ type: Input }],
    initialValue: [{ type: Input }],
    label: [{ type: Input }],
    name: [{ type: Input }],
    placeholder: [{ type: Input }],
    required: [{ type: Input }],
    type: [{ type: Input }],
    labelHidden: [{ type: Input }],
    hasError: [{ type: Input }],
    describedBy: [{ type: Input }],
    display: [{ type: HostBinding, args: ['style.display',] }]
};
