// @aws-amplify/ui-angular/esm2015/lib/components/authenticator/components/federated-sign-in-button/federated-sign-in-button.component.js
import { Component, Input } from '@angular/core';
import { AuthenticatorService } from '../../../../services/authenticator.service';
export class FederatedSignInButtonComponent {
    constructor(authenticator) {
        this.authenticator = authenticator;
        this.onClick = () => {
            this.authenticator.send({
                type: 'FEDERATED_SIGN_IN',
                data: { provider: this.provider },
            });
        };
    }
}
FederatedSignInButtonComponent.decorators = [
    { type: Component, args: [{
                selector: 'amplify-federated-sign-in-button',
                template: "<button\n  amplify-button\n  class=\"amplify-field-group__control federated-sign-in-button\"\n  fullWidth=\"true\"\n  fontWeight=\"normal\"\n  (click)=\"onClick()\"\n>\n  <div class=\"amplify-flex federated-sign-in-button-row\">\n    <ng-content></ng-content>\n  </div>\n</button>\n"
            },] }
];
FederatedSignInButtonComponent.ctorParameters = () => [
    { type: AuthenticatorService }
];
FederatedSignInButtonComponent.propDecorators = {
    provider: [{ type: Input }],
    text: [{ type: Input }]
};
