// @aws-amplify/ui-angular/esm2015/lib/primitives/select/select.component.js
import { Component, Input } from '@angular/core';
export class SelectComponent {
}
SelectComponent.decorators = [
    { type: Component, args: [{
                selector: 'amplify-form-select',
                template: "<label class=\"amplify-label amplify-visually-hidden\" [for]=\"id\">{{\n  label\n}}</label>\n<div class=\"amplify-select__wrapper\">\n  <select\n    class=\"amplify-select amplify-field-group__control\"\n    autocomplete=\"tel-country-code\"\n    [id]=\"id\"\n    [name]=\"name\"\n  >\n    <option\n      *ngFor=\"let item of items\"\n      [value]=\"item\"\n      [selected]=\"item === defaultValue\"\n    >\n      {{ item }}\n    </option>\n  </select>\n  <div\n    class=\"amplify-flex amplify-select__icon-wrapper\"\n    style=\"align-items: center; justify-content: center\"\n  >\n    <svg\n      xmlns=\"http://www.w3.org/2000/svg\"\n      class=\"amplify-icon amplify-icon--large\"\n      viewBox=\"0 0 24 24\"\n      data-size=\"large\"\n      fill=\"currentColor\"\n    >\n      <path d=\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\"></path>\n    </svg>\n  </div>\n</div>\n"
            },] }
];
SelectComponent.propDecorators = {
    items: [{ type: Input }],
    name: [{ type: Input }],
    label: [{ type: Input }],
    id: [{ type: Input }],
    defaultValue: [{ type: Input }]
};
