// @aws-amplify/ui-angular/esm2015/lib/primitives/button/button.component.js
import { Component, HostBinding, Input } from '@angular/core';
export class ButtonComponent {
    constructor() {
        this.type = 'button';
        this.fullWidth = false;
        this.isDisabled = false;
        this.size = 'medium';
        this.variation = 'default';
        this.fontWeight = 'normal';
    }
    get getDisabled() {
        return this.isDisabled ? '' : null;
    }
    get classNames() {
        let className = 'amplify-button';
        if (this.variation) {
            className += ` amplify-button--${this.variation}`;
        }
        if (this.size) {
            className += ` amplify-button--${this.size}`;
        }
        if (this.fullWidth) {
            className += ` amplify-button--fullwidth`;
        }
        if (this.isDisabled) {
            className += ` amplify-button--disabled amplify-button--loading`;
        }
        return className;
    }
    ngOnInit() {
        this.typeAttr = this.type;
        this.fullWidthAttr = this.fullWidth;
        this.sizeAttr = this.size;
        this.variationAttr = this.variation;
        this.fontWeightAttr = this.fontWeight;
    }
}
ButtonComponent.decorators = [
    { type: Component, args: [{
                selector: 'button[amplify-button]',
                template: "<ng-content></ng-content>\n"
            },] }
];
ButtonComponent.propDecorators = {
    type: [{ type: Input }],
    fullWidth: [{ type: Input }],
    isDisabled: [{ type: Input }],
    size: [{ type: Input }],
    variation: [{ type: Input }],
    fontWeight: [{ type: Input }],
    typeAttr: [{ type: HostBinding, args: ['type',] }],
    fullWidthAttr: [{ type: HostBinding, args: ['attr.data-fullwidth',] }],
    sizeAttr: [{ type: HostBinding, args: ['attr.data-size',] }],
    variationAttr: [{ type: HostBinding, args: ['attr.data-variation',] }],
    fontWeightAttr: [{ type: HostBinding, args: ['style.font-weight',] }],
    getDisabled: [{ type: HostBinding, args: ['attr.disabled',] }],
    classNames: [{ type: HostBinding, args: ['class',] }]
};
