// @aws-amplify/ui-angular/esm2015/lib/components/authenticator/components/base-form-fields/base-form-fields.component.js
import { Component, HostBinding, Input } from '@angular/core';
import { hasTranslation, translate, getDefaultFormFields, getActorContext, getSortedFormFields, applyDefaults, sortFormFields, } from '@aws-amplify/ui';
import { AuthenticatorService } from '../../../../services/authenticator.service';
/**
 * Sorts the given formFields, then renders them in order.
 */
export class BaseFormFieldsComponent {
    constructor(authenticator) {
        this.authenticator = authenticator;
        this.display = 'contents';
        this.formFields = [];
    }
    ngOnInit() {
        const state = this.authenticator.authState;
        this.formFields = getSortedFormFields(this.route, state);
        if (this.route === 'confirmSignUp') {
            this.handleConfirmSignUp();
        }
    }
    handleConfirmSignUp() {
        /**
         * @todo(breaking): Angular `confirmSignUp` has different placholder here from other frameworks.
         *
         * Translating here in a backwards-compatible manner, but should be resolved in next major version.
         */
        var _a;
        const state = this.authenticator.authState;
        // backwards compatible placeholder text
        const placeholder = !hasTranslation('Confirmation Code')
            ? translate('Enter your code') // prioritize new placeholder
            : translate('Confirmation Code'); // legacy placeholder
        let defaultFormFields = getDefaultFormFields(this.route, state);
        if (defaultFormFields.confirmation_code.placeholder) {
            defaultFormFields.confirmation_code.placeholder = placeholder;
        }
        const customFormFields = ((_a = getActorContext(state).formFields) === null || _a === void 0 ? void 0 : _a.confirmSignUp) || {};
        const newFormFields = applyDefaults(defaultFormFields, customFormFields);
        this.formFields = sortFormFields(newFormFields);
    }
}
BaseFormFieldsComponent.decorators = [
    { type: Component, args: [{
                selector: 'amplify-base-form-fields',
                template: "<ng-container *ngFor=\"let formField of formFields\">\n  <amplify-form-field\n    [name]=\"formField[0]\"\n    [formField]=\"formField[1]\"\n  ></amplify-form-field>\n</ng-container>\n"
            },] }
];
BaseFormFieldsComponent.ctorParameters = () => [
    { type: AuthenticatorService }
];
BaseFormFieldsComponent.propDecorators = {
    route: [{ type: Input }],
    display: [{ type: HostBinding, args: ['style.display',] }]
};
