// @aws-amplify/ui-angular/esm2015/lib/components/authenticator/components/confirm-sign-up/confirm-sign-up.component.js
import { Component, HostBinding, Input } from '@angular/core';
import { AuthenticatorService } from '../../../../services/authenticator.service';
import { getFormDataFromEvent, translate, } from '@aws-amplify/ui';
export class ConfirmSignUpComponent {
    constructor(authenticator) {
        this.authenticator = authenticator;
        this.headerText = translate('Confirm Sign Up');
        this.dataAttr = '';
        // translated texts
        this.resendCodeText = translate('Resend Code');
        this.confirmText = translate('Confirm');
        this.emailMessage = translate('Your code is on the way. To log in, enter the code we emailed to');
        this.textedMessage = translate('Your code is on the way. To log in, enter the code we texted to');
        this.defaultMessage = translate('Your code is on the way. To log in, enter the code we sent you. It may take a minute to arrive.');
        this.minutesMessage = translate('It may take a minute to arrive.');
    }
    get context() {
        return this.authenticator.slotContext;
    }
    get confirmSignUpHeading() {
        const { codeDeliveryDetails: { DeliveryMedium } = {} } = this.authenticator;
        return DeliveryMedium === 'EMAIL'
            ? translate('We Emailed You')
            : DeliveryMedium === 'SMS'
                ? translate('We Texted You')
                : translate('We Sent A Code');
    }
    get subtitleText() {
        const { codeDeliveryDetails: { DeliveryMedium, Destination } = {} } = this.authenticator;
        return DeliveryMedium === 'EMAIL'
            ? `${this.emailMessage} ${Destination}. ${this.minutesMessage}`
            : DeliveryMedium === 'SMS'
                ? `${this.textedMessage} ${Destination}. ${this.minutesMessage}`
                : translate(`${this.defaultMessage}`);
    }
    onInput(event) {
        event.preventDefault();
        const { name, value } = event.target;
        this.authenticator.updateForm({ name, value });
    }
    onSubmit(event) {
        event.preventDefault();
        this.authenticator.submitForm(getFormDataFromEvent(event));
    }
}
ConfirmSignUpComponent.decorators = [
    { type: Component, args: [{
                selector: 'amplify-confirm-sign-up',
                template: "<ng-container>\n  <form data-amplify-form (submit)=\"onSubmit($event)\" (input)=\"onInput($event)\">\n    <fieldset\n      class=\"amplify-flex amplify-authenticator__column\"\n      data-amplify-fieldset\n      [disabled]=\"context.isPending\"\n    >\n      <amplify-slot name=\"confirm-sign-up-header\" [context]=\"context\">\n        <h3\n          class=\"amplify-heading amplify-authenticator__heading\"\n          style=\"font-size: 1.5rem\"\n        >\n          {{ confirmSignUpHeading }}\n        </h3>\n      </amplify-slot>\n      <span class=\"amplify-authenticator__subtitle\">\n        {{ subtitleText }}\n      </span>\n      <amplify-base-form-fields\n        route=\"confirmSignUp\"\n      ></amplify-base-form-fields>\n\n      <button\n        amplify-button\n        variation=\"primary\"\n        fullWidth=\"true\"\n        type=\"submit\"\n        [isDisabled]=\"authenticator.isPending\"\n      >\n        {{ confirmText }}\n      </button>\n      <button\n        amplify-button\n        fontWeight=\"normal\"\n        (click)=\"authenticator.resendCode()\"\n      >\n        {{ resendCodeText }}\n      </button>\n    </fieldset>\n\n    <amplify-error *ngIf=\"context.error\">\n      {{ authenticator.error }}\n    </amplify-error>\n    <amplify-slot\n      name=\"confirm-sign-up-footer\"\n      [context]=\"context\"\n    ></amplify-slot>\n  </form>\n</ng-container>\n"
            },] }
];
ConfirmSignUpComponent.ctorParameters = () => [
    { type: AuthenticatorService }
];
ConfirmSignUpComponent.propDecorators = {
    headerText: [{ type: Input }],
    dataAttr: [{ type: HostBinding, args: ['attr.data-amplify-authenticator-confirmsignup',] }]
};
