// @aws-amplify/ui-angular/esm2015/lib/components/authenticator/components/sign-in/sign-in.component.js
import { Component, HostBinding, ViewEncapsulation, } from '@angular/core';
import { AuthenticatorService } from '../../../../services/authenticator.service';
import { translate, hasTranslation, getFormDataFromEvent, } from '@aws-amplify/ui';
export class SignInComponent {
    constructor(authenticator) {
        this.authenticator = authenticator;
        this.dataAttr = '';
        // translated phrases
        // Support backwards compatibility for legacy key with trailing space
        this.forgotPasswordText = !hasTranslation('Forgot your password? ')
            ? translate('Forgot your password?')
            : translate('Forgot your password? ');
        this.signInButtonText = translate('Sign in');
    }
    get context() {
        return this.authenticator.slotContext;
    }
    onInput(event) {
        event.preventDefault();
        const { name, value } = event.target;
        this.authenticator.updateForm({ name, value });
    }
    onSubmit(event) {
        event.preventDefault();
        this.authenticator.submitForm(getFormDataFromEvent(event));
    }
}
SignInComponent.decorators = [
    { type: Component, args: [{
                selector: 'amplify-sign-in',
                template: "<amplify-slot name=\"sign-in-header\" [context]=\"context\"></amplify-slot>\n\n<form data-amplify-form (submit)=\"onSubmit($event)\" (input)=\"onInput($event)\">\n  <amplify-federated-sign-in></amplify-federated-sign-in>\n  <fieldset\n    class=\"amplify-flex amplify-authenticator__column\"\n    style=\"flex-direction: column\"\n    data-amplify-fieldset\n    [disabled]=\"authenticator.isPending\"\n  >\n    <legend class=\"amplify-visually-hidden\">Sign in</legend>\n    <amplify-base-form-fields route=\"signIn\"></amplify-base-form-fields>\n    <button\n      amplify-button\n      variation=\"primary\"\n      fullWidth=\"true\"\n      type=\"submit\"\n      [isDisabled]=\"authenticator.isPending\"\n    >\n      {{ signInButtonText }}\n    </button>\n\n    <amplify-error *ngIf=\"authenticator.error\">\n      {{ authenticator.error }}\n    </amplify-error>\n  </fieldset>\n</form>\n\n<amplify-slot name=\"sign-in-footer\" [context]=\"context\">\n  <div data-amplify-footer>\n    <button\n      amplify-button\n      fontWeight=\"normal\"\n      size=\"small\"\n      variation=\"link\"\n      fullWidth=\"true\"\n      (click)=\"authenticator.toResetPassword()\"\n    >\n      {{ forgotPasswordText }}\n    </button>\n  </div>\n</amplify-slot>\n",
                encapsulation: ViewEncapsulation.None
            },] }
];
SignInComponent.ctorParameters = () => [
    { type: AuthenticatorService }
];
SignInComponent.propDecorators = {
    dataAttr: [{ type: HostBinding, args: ['attr.data-amplify-authenticator-signin',] }]
};
