// @aws-amplify/ui-angular/esm2015/lib/utilities/amplify-slot/amplify-slot.component.js
import { Component, HostBinding, Input, } from '@angular/core';
import { CustomComponentsService } from '../../services/custom-components.service';
export class AmplifySlotComponent {
    constructor(propService) {
        this.propService = propService;
        this.display = 'contents';
        this.isOverriden = false;
    }
    ngAfterContentInit() {
        const customComponents = this.propService.customComponents;
        const overridingComponent = customComponents[this.name];
        if (overridingComponent) {
            this.overridingComponent = overridingComponent;
            this.isOverriden = true;
        }
    }
}
AmplifySlotComponent.decorators = [
    { type: Component, args: [{
                selector: 'amplify-slot',
                template: "<!-- \n  if slot isn't overwritten, we display the the default coponent, which is the \n  children passed onto this component.\n-->\n<ng-content *ngIf=\"!isOverriden\"></ng-content>\n\n<!-- If slot is overwritten, we render that instead. -->\n<ng-container\n  *ngIf=\"isOverriden\"\n  [ngTemplateOutlet]=\"overridingComponent\"\n  [ngTemplateOutletContext]=\"context\"\n></ng-container>\n"
            },] }
];
AmplifySlotComponent.ctorParameters = () => [
    { type: CustomComponentsService }
];
AmplifySlotComponent.propDecorators = {
    name: [{ type: Input }],
    context: [{ type: Input }],
    display: [{ type: HostBinding, args: ['style.display',] }]
};
